File: engine/engine-model/src/main/scala/tools/packet/buddy/UpdateBuddyChannel.scala
package tools.packet.buddy

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class UpdateBuddyChannel(private var _characterId: Int, private var _channel: Int) extends PacketInput {
  def characterId: Int = _characterId

  def channel: Int = _channel

  override def opcode(): SendOpcode = SendOpcode.BUDDY_LIST
}

File: engine/engine-model/src/main/java/tools/packet/buddy/UpdateBuddyChannel.java
package tools.packet.buddy;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record UpdateBuddyChannel(int characterId, int channel) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.BUDDY_LIST;
   }
}
